### PR TITLE
Hughsie/udev tests prep2

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-udev-device.c [hunk 1]
```diff
@@ -1385,34 +1287,9 @@ fu_udev_device_open(FuDevice *device, GError **error)
...
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_INTERNAL,
-			    "rescan could not find device %s",
-			    sysfs_path);
-		return FALSE;
-	}
-	fu_udev_device_set_dev(self, udev_device);
+	if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_EMULATED))
+		return TRUE;
```
Comment (superm1) [inline libfwupdplugin/fu-udev-device.c:1291]: Hmm, so what do you want to do about emulation as it pertains to a sysfs file changing as part of a flash process?

**Review by superm1 — APPROVED** [review]
I'm fine with this set of changes.

File: libfwupdplugin/fu-udev-device.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:1291]: This made my head spin a little at first too -- essentially the old code was saying "the GUdevDevice that backs the FuUdevDevice might have changed" as GUdevDevice is a mostly immutable snapshot of a udev event. Now that FuUdevEvent actually models the sysfs device itself it no longer has to be invalidated. The contents of the sysfs might have changed (hence why we invalidate the probe data), but for emulations that's never going to happen (as it's the same device)  and we don't actually want to replay the probe out of order.

**Comment by d4s** [comment]
Just discovered, looks like the 296b8ac2bcd8b593f609c25e25dd5623d00d5de6 broke the logitech-hidpp plugin at least -- Bolt device doesn't detected anymore.
On it.

**Pull request merged**